cmd/signature-v2.go

```diff
@@ -124,55 +125,63 @@ func doesPresignV2SignatureMatch(r *http.Request) APIErrorCode {
 	// Extract the necessary values from presigned query, construct a list of new filtered queries.
 	for _, query := range unescapedQueries {
 		keyval := strings.SplitN(query, "=", 2)
+		if len(keyval) != 2 {
```

Comment:
Good catch!

Reply:
Thanks 😃 
Without the check you can cause panics if you send pre-signed requests without e.g. a signature.
Luckily the server survives the panic - it only kills the go routine. Otherwise we would have an additional DoS vul.